manticore/platforms/linux.py

```diff
@@ -383,11 +393,18 @@ def _execve(self, program, argv, envp):
             self.forward_events_from(proc)
 
     def _mk_proc(self, arch):
-        if arch in {'i386', 'armv7'}:
-            mem = Memory32()
-        else:
-            mem = Memory64()
-        return CpuFactory.get_cpu(mem, arch)
+        mem = Memory32() if arch in {'i386', 'armv7'} else Memory64()
+        cpu = CpuFactory.get_cpu(mem, arch)
+        # FIXME
+        arch_map = {
+            'i386': (cs.CS_ARCH_X86, cs.CS_MODE_32),
+            'amd64': (cs.CS_ARCH_X86, cs.CS_MODE_64),
+            'armv7': (cs.CS_ARCH_ARM, cs.CS_MODE_ARM)
+        }
+        arch, mode = arch_map[arch]
+        cpu.__class__.disasm = init_disassembler(self.disasm, arch, mode, self.programs)
```

Comment:
The disassembler passed in the CPU is no longer a string, but a class instance which is passed from the platform. Initially I believe I had it in the constructor, the problem is that all current tests don't know about this so the test should be updated to also initialize a disassembler via the `init_disassembler` call, before creating the CPU. Also you mean adding it to the Cpu class or each of the subclasses?